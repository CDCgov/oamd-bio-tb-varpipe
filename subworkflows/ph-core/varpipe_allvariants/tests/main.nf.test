nextflow_workflow {
    name "Test Subworkflow VARPIPE_ALLVARIANTS"
    script "../main.nf"
    workflow "VARPIPE_ALLVARIANTS"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_phcore"
    tag "subworkflows/varpipe_allvariants"
    tag "gatk4/leftalignandtrimvariants"
    tag "gatk4/mutect2"
    tag "gatk4/filtermutectcalls"
    tag "varpipe/snpeff"

    test("varpipe-allvariants subworkflow test") {
        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                // Prepare input channels
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + "/genomics/tuberculosis/bam/test.bam", checkIfExists: true),
                        file(params.modules_testdata_base_path + "/genomics/tuberculosis/bam/test.bai", checkIfExists: true),
                        []
                    ]
                )

                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + "/genomics/tuberculosis/genome/reference-index/ref2.fa.fai", checkIfExists: true)
                    ]
                )

                input[2] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + "/genomics/tuberculosis/genome/reference-index/ref2.dict", checkIfExists: true)
                    ]
                )

                input[3] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + "/genomics/tuberculosis/genome/kb/varpipe/ref2.fa", checkIfExists: true)
                    ]
                )
                input[4] = 'NC_000962'
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.ann.size() > 0 },
                { assert snapshot(
                        workflow.out.versions
                    ).match()
                }
            )
        }
    }

}
