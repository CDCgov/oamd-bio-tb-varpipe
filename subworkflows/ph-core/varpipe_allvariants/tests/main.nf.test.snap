{
    "varpipe-allvariants subworkflow test": {
        "content": [
            [
                "versions.yml:md5,50aca57d44b516f9fd6f3c979c80e805",
                "versions.yml:md5,697fbc3ceb04999ef1d7cd812cb317bc",
                "versions.yml:md5,a6319944a4de0085a9ecf40478968360",
                "versions.yml:md5,b84285b031458bee8a2d9972373f396b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-16T11:14:29.352682733"
    }
}
